mentor-match-app/src/components/profile/LoggedUserProfile.jsx

```diff
@@ -0,0 +1,490 @@
+                      <Stack spacing={1} width={'65%'}>
+                        <Stack
+                          direction="row"
+                          justifyContent={'space-between'}
+                          alignItems={'center'}
+                        >
+                          <Typography variant={'h6'} fontWeight={'regular'}>
+                            {user.displayName}
+                          </Typography>
+
+                          {user.role && (
+                            <Stack direction="row" spacing={0.5}>
+                              <Typography
+                                variant={'body2'}
+                                color="text.secondary"
+                              >
+                                {user.role}
+                              </Typography>
+                              <SchoolIcon fontSize="small" color="secondary" />
+                            </Stack>
+                          )}
+                        </Stack>
+                        <Divider orientation="horizontal" />
+                        {user.profileDescription && (
+                          <Typography
+                            variant={'body2'}
+                            color="text.secondary"
+                            py={1.5}
+                          >
+                            {user.profileDescription}
+                          </Typography>
+                        )}
+                        <Typography variant={'body2'} lineHeight={2}>
+                          {user.title && (
+                            <Stack
+                              direction="row"
+                              spacing={0.5}
+                              alignItems="center"
+                            >
+                              <PersonIcon fontSize="small" color="primary" />
+                              <span>{user.title}</span>
+                            </Stack>
+                          )}
+                          {user.affiliation && (
+                            <Stack
+                              direction="row"
+                              spacing={0.5}
+                              alignItems="center"
+                            >
+                              <BusinessIcon fontSize="small" color="primary" />
+                              <span>Affiliation: {user.affiliation}</span>
+                            </Stack>
+                          )}
+                          {user.location && (
+                            <Stack
+                              direction="row"
+                              spacing={0.5}
+                              alignItems="center"
+                            >
+                              <LocationIcon fontSize="small" color="primary" />
+                              <span>Location: {user.location}</span>
+                            </Stack>
+                          )}
+
+                          {formAnswers && (
```

Comment:
Add Better Object Access Safety
```suggestion
                         {formAnswers && (
+   Object.keys(formAnswers).length > 0 && (
    <Stack direction="row" spacing={0.5} alignItems="center">
      <LanguageIcon fontSize="small" color="primary" />
      <span>
        Languages:{' '}
        {formAnswers.mentorData
-         ? formAnswers.mentorData.languages.join(', ')
-         : formAnswers.menteeData.languages.join(', ')}
+         ? formAnswers.mentorData.languages?.join(', ')
+         : formAnswers.menteeData?.languages?.join(', ')}
      </span>
    </Stack>
+   )
  )}
```